File: tests/corpus/clean/prompts.py
"""Prompt construction for OpsBot (safe)."""

from langchain.prompts import ChatPromptTemplate

# Prompt-named variable, matches BAS-LLM08-001's name constraint -- but no
# sk-shaped substring anywhere in the text, so the content constraint never
# fires. Name alone is never sufficient.
SYSTEM_PROMPT = "You are OpsBot, an internal infrastructure assistant. Help the engineer triage the incident described below."

# Proper delimiters: the untrusted value sits in its own templated slot via
# LangChain's message list, never string-interpolated into the instruction
# text. Not an f-string at all, so BAS-ZT4-001 cannot structurally match it.
chat_prompt = ChatPromptTemplate.from_messages(
    [("system", SYSTEM_PROMPT), ("human", "{sanitized_incident_summary}")]
)

File: tests/corpus/clean/prompts.ts
/**
 * Prompt construction for OpsBot's Node service (safe).
 */

import { ChatPromptTemplate } from "@langchain/core/prompts";

// Prompt-named variable -- no sk--shaped content anywhere in the text.
export const SYSTEM_PROMPT =
  "You are OpsBot, an internal infrastructure assistant. Help the engineer triage the incident described below.";

// Proper delimiters: the untrusted value sits in its own templated slot,
// never string-interpolated into the instruction text.
export const chatPrompt = ChatPromptTemplate.fromMessages([
  ["system", SYSTEM_PROMPT],
  ["human", "{sanitizedIncidentSummary}"],
]);
